openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -659,3 +662,65 @@ def password_reset_token_validate(request):
         AUDIT_LOG.exception("Invalid password reset confirm token")
 
     return JsonResponse({'is_valid': is_valid})
+
+
+def _check_token_has_required_values(**kwargs):
+    """
+    Helper function to test that token
+    string passed has the required kwargs needed
+    to process token validation.
+    """
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not uidb36 or not token:
+        return False
+    try:
+        _ = base36_to_int(uidb36)
+    except ValueError:
+        return False
+    return True
+
+
+@require_POST
+@ensure_csrf_cookie
+def password_reset_logistration(request, **kwargs):
+    """Reset learner password using passed token and new credentials"""
+
+    reset_status = False
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not _check_token_has_required_values(**kwargs):
+        AUDIT_LOG.exception("Invalid password reset confirm token")
+        return JsonResponse({'reset_status': reset_status})
+
+    request.POST = request.POST.copy()
+    request.POST['new_password1'] = normalize_password(request.POST['new_password1'])
+    request.POST['new_password2'] = normalize_password(request.POST['new_password2'])
+
+    password = request.POST['new_password1']
+    try:
+        uid_int = base36_to_int(uidb36)
+        user = User.objects.get(id=uid_int)
+        if not default_token_generator.check_token(user, token):
```

Comment:
Do we also need to check if the logged in user is different from the token user?

Reply:
Any update on this ^^?

Reply:
No need to do this.

Reply:
Could you please explain a reason why this isn't required?

Reply:
First this link could be accessed with Anonymous session. So whosoever has the correct link (acting as key and should be protected) generated against an email address should have access to password reset functionality. The checks in some views were testing if the logged in user is different from the passed uid, meaning logged in user is clicking on someone link but the user can always logged out of the system to bypass this check by becoming Anonymous. So my understanding is even with adding this check, user would have a way around if he/she has the correct uid-token string.